doc/continuous-integration.md

```diff
@@ -0,0 +1,33 @@
+# Continuous integration
+
+## Main
+
+The [govuk_smart_answers CI instance](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/) is triggered by new commits on the `master` branch of the [GitHub repo](https://github.com/alphagov/smart-answers). It runs the [`jenkins.sh`](https://github.com/alphagov/smart-answers/blob/master/jenkins.sh) script with the `RUN_REGRESSION_TESTS` not set. This script does the following:
+
+* Checks out the lastest [govuk-content-schemas](https://github.com/alphagov/govuk-content-schemas) which are used in a few of the tests
+* Installs the bundled gems
+* Runs govuk-lint-ruby (see [Rubocop docs](doc/rubocop.md))
+* Runs the `test` rake task with `TEST_COVERAGE` enabled
+* Pre-compiles assets
+
+If any of these steps fails, the build fails-fast.
+
+## Branches
+
+The [govuk_smartanswers_branches CI instance](https://ci-new.alphagov.co.uk/job/govuk_smartanswers_branches/) is triggered by new commits on other branches of the [GitHub repo](https://github.com/alphagov/smart-answers). It runs the [`jenkins_branches.sh`](https://github.com/alphagov/smart-answers/blob/master/jenkins_branches.sh) script with the `RUN_REGRESSION_TESTS` not set.
+
+This script does the same as the [main CI instance](#main), but first it tries to merge `master` into the relevant branch in order to work out whether it will merge without conflicts. This merge is always aborted, but if conflicts are detected the script fails fast at this point.
```

Comment:
1. I meant "the script" in the sense of the combined behaviour of running the `jenkins_branches.sh` script, but I see that that's not obvious.

2. Hmm. I think the command `git merge --abort` is always executed (whether or not `git merge --no-commit origin/master` succeeds or fails), but perhaps the abort has no effect if the merge succeeds. It makes me wonder whether the intention was that the merge should be reverted!

I'll update the docs based on your feedback. Thanks.